wcivf/apps/feedback/templates/feedback/feedback_form.html

```diff
@@ -19,24 +22,104 @@ <h5>
             <span aria-hidden="true">🔎</span>
             {% trans "Did you find what you were looking for?" %}
         </h5>
-        {% for choice in feedback_form.found_useful %}
-            {{ choice.tag }}
-            <label class="link-button" for="{{ choice.id_for_label }}">{{ choice.choice_label }}</label>
-        {% endfor %}
-
+        <div id="feedback_choices_container">
+            {% for choice in feedback_form.found_useful %}
+                {{ choice.tag }}
+                <label class="link-button" for="{{ choice.id_for_label }}">{{ choice.choice_label }}</label>
+            {% endfor %}
+        </div>
         <h5>
             <span aria-hidden="true">🗳️</span>
             {% trans "Has this information made you more likely to vote?" %}
         </h5>
-        {% for choice in feedback_form.vote %}
-            {{ choice.tag }}
-            <label class="link-button" for="{{ choice.id_for_label }}">{{ choice.choice_label }}</label>
-        {% endfor %}
+        <div id="voting_choices_container">
+            {% for choice in feedback_form.vote %}
+                {{ choice.tag }}
+                <label class="link-button" for="{{ choice.id_for_label }}">{{ choice.choice_label }}</label>
+            {% endfor %}
+        </div>
 
-        <div>
+        <div id="comments">
             <p>{% trans "Can you tell us anything more?" %}</p>
             {{ feedback_form.comments }}
         </div>
         <button type="submit" class="ds-button-pink">{% trans "Send feedback" %}</button>
     </form>
 </div>
+
+{% block in_page_javascript %}
+    <script>
+        document.addEventListener("DOMContentLoaded", function(event) {
+            var feedback_form = document.forms[1]
+            var feedback_form_token = feedback_form.elements[1].value
+            var csrfmiddlewaretoken = feedback_form.elements[0].value
+            var found_useful_label = feedback_form.elements[3].labels[0]
+            // TO DO: the label should not be active on page load
+
+
+            var feedback_choices_container = document.getElementById('feedback_choices_container')
+            var voting_choices_container = document.getElementById('voting_choices_container')
+            var feedback_labels = feedback_choices_container.getElementsByTagName('label')
+            var voting_labels = voting_choices_container.getElementsByTagName('label')
+
+            /// Remove the comments box until the user has selected 'YES' or 'NO'
+            var found_useful = feedback_form.elements[3].value
+
+            var comments = document.getElementById('comments')
+            comments.hidden = true
+
+            var last_feedback_choice = document.getElementById('feedback_choices_container').lastElementChild
+            var problem_radio = '<input type="radio" name="report_problem" value="PROBLEM" required data-toggle="button" id="report_problem" />'
```

Comment:
I'm not really sure what this code is doing. If it's showing / hiding form elements, then can you write the HTML out in the template and then just toggle a hidden flag on the elements? Making the form in JS feels like it might break something at some point.

Also, because the Django tests don't run JS, I think this will make testing harder. If the form works without JS enabled, we're able to write tests for it

Reply:
I overthought how this might done, fixed up now, thank you 👍

Reply:
The only remaining issues:

- test failure: submitting the form from `/feedback/` (which the user may never do since we don't link to it from anywhere) does not create a `Feedback` object
- active status of buttons when they are not selected. I've tried playing around with removing styles and making button links inactive, but have not figured this out yet.